### ISSUE TITLE
MM 0.47.6 - game hangs when indirect fire in flight and firing unit destroyed

### ISSUE THREAD

**Issue by nsecomb** [MegaMek/megamek#2023]
# Environment
HQ's MegaMek 0.47.6 [_Addendum: have also run and replicated same behaviour in HQ's MegaMek 0.47.7_]
Windows 10 Version 2004 (OS Build 190941.329)
Java(TM) SE Runtime Environment (build 1.8.0_251-b08)
Java HotSpot(TM) 64-Bit Server VM (build 25.251-b08, mixed mode)

### Description
Game hangs when indirect arrow IV rounds in flight and firing unit has been destroyed.

**Base parameters**: 2 player units, 1 bot unit, 20x60 random map, 0.82 gravity, Full-moon night, indirect rounds fired (illumination, > 20 hexes for 1 round flight time), mostly default settings except paranoid autosave = on, turn timer = 2 mins

**Test Case 1**: No indirect rounds in flight when firing unit destroyed/ejected
Unit = Catapult
Outcome: game completed normally.

**Test Case 2**: Indirect rounds in flight when firing unit destroyed/ejected (keep firing indirect rounds at 20+ hexes until unit destroyed)
Unit = Regulator (Arrow IV)
Outcome: game hangs, possibly when trying to enter indirect fire phase

### Files

playername: NigeTest   (no password)

File sets: 
**Test Case 1**: 
Indirect_fire_bug_catapult_lobby.sav.gz : save file from lobby before entering game
Indirect_fire_bug_catapult_finish.sav.gz :  save file at end of game
megameklog_reproduction_catapult_ok.txt : log file after successful game completion

**Test Case 2**: 
Indirect_fire_bug_regulator_lobby.sav.gz : save file from lobby before entering game
Indirect_fire_bug_regulator_start.sav.gz :  save file at post-deployment
Indirect_fire_bug_regulator_destroyed.sav.gz :  save file at destruction of indirect firing unit while rounds still in air, and before game hangs
megameklog_reproduction_regulator_hung.txt : log file after game has hung

**Addendum**:  Test Case 2 replication in 0.47.7
Indirect_fire_bug_47_7_regulator_lobby.sav.gz : save file from lobby before entering game
Indirect_fire_bug_47_7_regulator_start.sav.gz : save file at post-deployment
Indirect_fire_bug_47_7_regulator_destroyed.sav.gz : save file at destruction of indirect firing unit while rounds still in air, and before game hangs
megameklog_reproduction_47_7_regulator_hung.txt : log file after game has hung

[Indirect_fire_bug_catapult_lobby.sav.gz](https://github.com/MegaMek/megamek/files/4822860/Indirect_fire_bug_catapult_lobby.sav.gz)
[Indirect_fire_bug_catapult_finish.sav.gz](https://github.com/MegaMek/megamek/files/4822862/Indirect_fire_bug_catapult_finish.sav.gz)
[Indirect_fire_bug_regulator_lobby.sav.gz](https://github.com/MegaMek/megamek/files/4822863/Indirect_fire_bug_regulator_lobby.sav.gz)
[Indirect_fire_bug_regulator_start.sav.gz](https://github.com/MegaMek/megamek/files/4822864/Indirect_fire_bug_regulator_start.sav.gz)
[Indirect_fire_bug_regulator_destroyed.sav.gz](https://github.com/MegaMek/megamek/files/4822867/Indirect_fire_bug_regulator_destroyed.sav.gz)
[megameklog_reproduction_catapult_ok.txt](https://github.com/MegaMek/megamek/files/4822871/megameklog_reproduction_catapult_ok.txt)
[megameklog_reproduction_regulator_hung.txt](https://github.com/MegaMek/megamek/files/4822872/megameklog_reproduction_regulator_hung.txt)

[Indirect_fire_bug_47_7_regulator_lobby.sav.gz](https://github.com/MegaMek/megamek/files/4823131/Indirect_fire_bug_47_7_regulator_lobby.sav.gz)
[Indirect_fire_bug_47_7_regulator_start.sav.gz](https://github.com/MegaMek/megamek/files/4823133/Indirect_fire_bug_47_7_regulator_start.sav.gz)
[Indirect_fire_bug_47_7_regulator_destroyed.sav.gz](https://github.com/MegaMek/megamek/files/4823134/Indirect_fire_bug_47_7_regulator_destroyed.sav.gz)
[megameklog_reproduction_47_7_regulator_hung.txt](https://github.com/MegaMek/megamek/files/4823142/megameklog_reproduction_47_7_regulator_hung.txt)

**Comment by NickAragua** [comment]
I hate to ask, but does this happen in the just-released 47.7?

**Comment by HammerGS** [comment]
NPE in Log

```
12:00:35,512 ERROR [megamek.server.Server] {Packet Pump} 
checkExtremeGravityMovement(Entity,MoveStep,EntityMovementType,Coords,int) : gravity move check jump: 6/6
Exception in thread "Packet Pump" java.lang.NullPointerException
	at megamek.common.weapons.ArtilleryWeaponIndirectFireHandler.handle(ArtilleryWeaponIndirectFireHandler.java:158)
	at megamek.server.Server.handleAttacks(Server.java:35820)
	at megamek.server.Server.handleAttacks(Server.java:35749)
	at megamek.server.Server.endCurrentPhase(Server.java:2994)
	at megamek.server.Server.changePhase(Server.java:2279)
	at megamek.server.Server.endCurrentPhase(Server.java:2900)
	at megamek.server.Server.changeToNextTurn(Server.java:3246)
	at megamek.server.Server.endCurrentTurn(Server.java:2255)
	at megamek.server.Server.receiveMovement(Server.java:5004)
	at megamek.server.Server.handle(Server.java:31687)
	at megamek.server.Server$PacketPump.run(Server.java:263)
```

**Comment by nsecomb** [comment]
> 
> 
> I hate to ask, but does this happen in the just-released 47.7?

Yes, just ran Test Case 2 in 0.47.7 and same behaviour. Will try and add the 0.47.7 files to ticket.

**Comment by NickAragua** [comment]
Ok, I should be able to track it down with that info. Thanks for the report!

**Comment by sixlettervariables** [comment]
We fixed the specific call site in 47.6 with 47.7. An updated error log from 47.7 can help too.

Also @nsecomb thank you for the very detailed report.
